File: experiments/006-SmalltalkVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260915-111749-40413/analysis/acceptance-project/src/main/java/surveycrawler/Boulder.java
package surveycrawler;

public class Boulder extends Ground {

    // moving

    @Override
    public void moveCrawlerTo(SurveyCrawler aCrawler, Point aNewPosition, Point aDirection) {
        aCrawler.signalCanNotMoveToBoulder();
    }

    // facing

    @Override
    public void turnCrawlerClockwise(SurveyCrawler aCrawler) {
        aCrawler.signalCanNotTurnOnBoulder();
    }

    @Override
    public void turnCrawlerCounterClockwise(SurveyCrawler aCrawler) {
        aCrawler.signalCanNotTurnOnBoulder();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-213927-34997/output/project/src/surveycrawler/boulder.py
from surveycrawler.ground import Ground


class Boulder(Ground):
    """The crawler can neither move to a boulder nor turn while standing on one."""

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        a_crawler.signal_can_not_move_over_boulder()

    # facing

    def turn_crawler_clockwise(self, a_crawler):
        a_crawler.signal_can_not_turn_over_boulder()

    def turn_crawler_counter_clockwise(self, a_crawler):
        a_crawler.signal_can_not_turn_over_boulder()
